### PR TITLE
[SYCL] Use multi_ptr to clean up deprecated warnings

### PR DESCRIPTION
**Pull request by AidanBeltonS**
This PR adds a `get_pointer()` helper function to use the more up-to-date `get_multi_ptr`.

This cleans up build warnings significantly.

- [x] I have read the [contributing guidelines](https://github.com/ggerganov/llama.cpp/blob/master/CONTRIBUTING.md)
- Self-reported review complexity:
  - [x] Low
  - [ ] Medium
  - [ ] High

### CODE REVIEW

**Comment by HanClinto** [comment]
We [just fixed](https://github.com/ggerganov/llama.cpp/pull/8258) the problems that editorconfig is [complaining about](https://github.com/ggerganov/llama.cpp/actions/runs/9762263887/job/26945279778?pr=8256), so if you update with a rebase or merge with master, you should be able to get a clean bill-of-health on the CI.

**Comment by airMeng** [comment]
Can you remote the other ```get_pointer``` too?
```
$:~/llama.cpp/ggml$ grep -rI "get_pointer()"
src/ggml-sycl.cpp:                                                                             local_buf_acc.get_pointer());
src/ggml-sycl/dpct/helper.hpp:            : accessor(acc.get_pointer(), in_range) {}
src/ggml-sycl/dpct/helper.hpp:            : accessor(acc.get_pointer(), in_range) {}
```

**Comment by AidanBeltonS** [comment]
> Can you remove the other `get_pointer` too?
> 
> ```
> $:~/llama.cpp/ggml$ grep -rI "get_pointer()"
> src/ggml-sycl.cpp:                                                                             local_buf_acc.get_pointer());
> src/ggml-sycl/dpct/helper.hpp:            : accessor(acc.get_pointer(), in_range) {}
> src/ggml-sycl/dpct/helper.hpp:            : accessor(acc.get_pointer(), in_range) {}
> ```

Thanks for pointing these out. I have addressed the `src/ggml-sycl.cpp` file.

I have not touched the dpct helpers, I am working on a separate PR to clean up some warnings in that. I will include the changes in the second PR, as I want to provide feedback to DPCT and provide an alternative that is independent of llama.cpp.

File: ggml/src/ggml-sycl/common.hpp [hunk 1]
```diff
@@ -351,4 +351,9 @@ static __dpct_inline__ float warp_reduce_max(float x,
     return x;
 }
 
+template <typename Tp, int dim = 1>
```
Comment (joeatodd) [inline ggml/src/ggml-sycl/common.hpp:354]: Nit: does `dim` need a default value?

**Review by joeatodd — APPROVED** [review]
LGTM - thanks @AidanBeltonS, it's a tidy solution

File: ggml/src/ggml-sycl/common.hpp (hunk 1 above)
Comment (AidanBeltonS) [reply ggml/src/ggml-sycl/common.hpp]: Done

**Review by OuadiElfarouki — APPROVED** [review]
Looks good!

**Pull request merged**